### PR TITLE
distro: add support for Flatcar Linux

### PR DESCRIPTION
**Pull request by ahjohannessen**
- make systemd unit path configurable as
   Flatcar Linux /lib/systemd/system is read-only.

### CODE REVIEW

**Comment by ahjohannessen** [comment]
@bbaassssiiee WDYT?

**Comment by bbaassssiiee** [comment]
ansible_python_interpreter can best be set in the inventory.
I see too many defines of control_host_ansible_python_interpreter

**Review by bbaassssiiee — COMMENTED** [review]
Please remove `control_host_ansible_python_interpreter`, you can set it in inventory.

**Comment by ahjohannessen** [comment]
>Please remove control_host_ansible_python_interpreter

@bbaassssiiee  `control_host_ansible_python_interpreter` is set because when host is e.g. MacOS the control host has path `/usr/bin/python` and the target host has, in my case, `/home/core/bin/python` and I have no way of defining a symbolic link to `/usr/bin/python` on Flatcar Linux as `/usr` is read-only.

The problem I am trying to solve with `control_host_ansible_python_interpreter` is that `ansible_python_interpreter` is different on the control host and target host(s) when using Flatcar Linux.

**Comment by ahjohannessen** [comment]
@bbaassssiiee I think I can solve it as you suggest pr. host. I will test and update the PR 👍

**Comment by ahjohannessen** [comment]
@bbaassssiiee I have updated the PR as you suggested! It works fine with overriding `ansible_python_interpreter` in inventory :)

**Comment by ahjohannessen** [comment]
@bbaassssiiee Is there anything else I can do in order to get this merged? :)

**Comment by bbaassssiiee** [comment]
I had to lookup Flatcar Linux, it is supposed to run containers. Why not run consul in a container on it?

**Comment by ahjohannessen** [comment]
> Why not run consul in a container on it?

Mostly because I can piggy-back on the conventions in this role and to avoid running consul itself in a container as it is not recommended by hashicorp.

**Review by bbaassssiiee — CHANGES_REQUESTED** [review]
Please fix `vars/Flatcar.yml` to pass yamllint in the GitHub checks.
Bonus: add a Flatcar scenario to **molecule**

**Comment by ahjohannessen** [comment]
@bbaassssiiee 

>Please fix vars/Flatcar.yml to pass yamllint in the GitHub checks.

Should be fixed now.

>Bonus: add a Flatcar scenario to molecule

I tried to see how the other scenarios are done, but could not find a suitable image for it.

File: vars/Flatcar.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+# File: Flatcar.yml - Flatcar variables for Consul
+
+consul_os_packages: []
+
+consul_systemd_unit_path: "/etc/systemd/system"
```
Comment (bbaassssiiee) [inline vars/Flatcar.yml]: Newline at EOF is missing.

**Review by bbaassssiiee — CHANGES_REQUESTED** [review]

**Comment by ahjohannessen** [comment]
@bbaassssiiee I think it should be ok now.

**Comment by ahjohannessen** [comment]
@bbaassssiiee I have added a list of supported *nix distributions in order to avoid lint error with max characters

**Comment by bbaassssiiee** [comment]
Nice!

**Review by bbaassssiiee — APPROVED** [review]

**Comment by ahjohannessen** [comment]
:+1:

**Pull request merged**